### PR TITLE
feat: add provider logs

### PR DESCRIPTION
**Pull request by voikin**
more details in `ASP-711`

### CODE REVIEW

File: object/provider.go [hunk 1]
```diff
@@ -261,21 +265,32 @@ func UpdateProvider(id string, provider *Provider) (bool, error) {
...
 	if provider.Type == "Tencent Cloud COS" {
 		provider.Endpoint = util.GetEndPoint(provider.Endpoint)
 		provider.IntranetEndpoint = util.GetEndPoint(provider.IntranetEndpoint)
 	}
 
 	affected, err := ormer.Engine.Insert(provider)
 	if err != nil {
+		logger.Error(ctx, "failed to add provider", "provider", provider.GetId())
 		return false, err
+	} else {
```
Comment (svgrafov) [inline object/provider.go:292]: You dont really need this "else" operator.
The code is equivalent to this clearer version using "return early" pattern
```
if err != nil {
    logger.Error(ctx, "failed to add provider", "provider", provider.GetId())
    return false, err
}

logger.Info(ctx, "successfully added provider", "provider", provider.GetId())
return affected != 0, nil
```

File: object/provider.go [hunk 2]
```diff
@@ -261,21 +265,32 @@ func UpdateProvider(id string, provider *Provider) (bool, error) {
...
+			logger.Info(ctx, "successfully updated provider mappings", "provider", provider.GetId())
+		}
+		logger.Info(ctx, "successfully updated provider", "provider", provider.GetId())
 	}
 
 	return affected != 0, nil
 }
 
 func AddProvider(provider *Provider) (bool, error) {
+	ctx := context.Background()
```
Comment (svgrafov) [inline object/provider.go:281]: Please use higher-level context instead.

**Review by svgrafov — CHANGES_REQUESTED** [review]
Feedback on some naming, test and context usage

File: object/provider_util.go [hunk 3]
```diff
@@ -8,3 +8,35 @@ func mapProvidersToNames(providers []*ProviderItem) []string {
 
 	return providerNames
 }
+
+func CompareRoleMappingSlices(slice1, slice2 []*RoleMappingItem) bool {
```
[the remark this answers is not in the document]
Comment (voikin) [reply object/provider_util.go]: done

File: object/provider.go (hunk 1 above)
Comment (voikin) [reply object/provider.go]: done

File: object/provider.go (hunk 2 above)
Comment (voikin) [reply object/provider.go]: done

**Review by svgrafov — APPROVED** [review]

**Review by dplynskiy — APPROVED** [review]

**Pull request merged**